osu.Game/Screens/Play/HUD/FaillingLayer.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    /// <summary>
+    /// An overlay layer on top of the playfield which fades to red when the current player health falls a certain threshold defined by <see cref="LowHealthThreshold"/>.
+    /// </summary>
+    public class FaillingLayer : HealthDisplay
+    {
+        private const float max_alpha = 0.4f;
+
+        private readonly Box box;
+
+        /// <summary>
+        /// The threshold under which the current player life should be considered low and the layer should start fading in.
+        /// </summary>
+        protected virtual double LowHealthThreshold => 0.20f;
```

Comment:
Not a fan of this being only-protected-get to be fair. Having to subclass just to change this seems senseless, and this component isn't really that reusable anyways (it's just a box with alpha, what's the point of conflating it with anything else?). I'd just make it a plain settable property.